DMCompiler/DM/Expressions/Constant.cs

```diff
@@ -83,6 +83,44 @@ abstract class Constant : DMExpression {
         public virtual Constant BinaryOr(Constant rhs) {
             throw new CompileErrorException(Location, $"const operation \"{this} | {rhs}\" is invalid");
         }
+
+        // Not a virtual because it'd be very annoying to write this for every Constant subtype
+        public Constant Equal(Constant rhs) {
+            switch (this) {
+                case Expressions.Null nil:
```

Comment:
#879?

Reply:
That line specifically is for `null`, not NaN.